### PR TITLE
chore: correct nuget dependencies

### LINKED ISSUE #330: Introduce package ranges to support all minors under a major
**Is your feature request related to a problem? Please describe.**
Currently, we fixate on a single version for a package dependency, which could lead to problems when users want to use Arcus libraries.

**Describe the solution you'd like**
Use package ranges for all minors under a major.

### PR DESCRIPTION
**Pull request by stijnmoreels**
Correct the NuGet dependencies to start from the current minor and take every minor that follows till the next major.
Includes reverts of Renovate on IoT and Serilog.Enrichers.

Closes #330

### CODE REVIEW

**Comment by stijnmoreels** [comment]
Wondering if this package ranges is better than just providing the lowest possible version (what we did before).
Looking at other libraries, they don't use ranges but a version (so translates to "this version or anything above").

File: src/Arcus.Observability.Correlation/Arcus.Observability.Correlation.csproj [hunk 1]
```diff
@@ -19,8 +19,8 @@
 
   <ItemGroup>
     <PackageReference Include="Guard.Net" Version="1.2.0" />
-    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="3.1.8" />
-    <PackageReference Include="Microsoft.Extensions.Options" Version="3.1.8" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="[3.1.8,5.0.0)" />
+    <PackageReference Include="Microsoft.Extensions.Options" Version="[3.1.8,5.0.0)" />
```
[the remark this answers is not in the document]
Comment (stijnmoreels) [reply src/Arcus.Observability.Correlation/Arcus.Observability.Correlation.csproj:23]: Yes, that's right. And 6.0.0 is targeting .NET 6.

[the remark this answers is not in the document]
Comment (fgheysels) [reply src/Arcus.Observability.Correlation/Arcus.Observability.Correlation.csproj:23]: If we're targetting .NET core 3.1, should we then allow that we can upgrade to version 5 of that package ?

[the remark this answers is not in the document]
Comment (fgheysels) [reply src/Arcus.Observability.Correlation/Arcus.Observability.Correlation.csproj:23]: I think we should apply wildcards on the max allowed version.  
Maybe we can just say
```
    <PackageReference Include="Microsoft.Extensions.Options" Version="3.*" />
```
Or something like that ?

[the remark this answers is not in the document]
Comment (stijnmoreels) [reply src/Arcus.Observability.Correlation/Arcus.Observability.Correlation.csproj:23]: > If we're targetting .NET core 3.1, should we then allow that we can upgrade to version 5 of that package ?

We don't allow that version 5 of that package with the version range here, `[3.1.8,5.0.0)` means >= 3.1.8 & < 5.0.0.

I think the only difference with the wildcard, is that we allow lower versions of 3.1.8. Hmm, yeah, maybe we should do that.

[the remark this answers is not in the document]
Comment (stijnmoreels) [reply src/Arcus.Observability.Correlation/Arcus.Observability.Correlation.csproj:23]: If it's mostly the higher version we should guard against, we can do that with ranges, if we want to take in all lower minors too, we can do a wildcard. Preference?

[the remark this answers is not in the document]
Comment (stijnmoreels) [reply src/Arcus.Observability.Correlation/Arcus.Observability.Correlation.csproj]: Discussed online. Changed as described https://github.com/arcus-azure/arcus/issues/188

**Pull request merged**